cmake/HiPACEFunctions.cmake

```diff
@@ -72,13 +72,6 @@ macro(set_default_build_type default_build_type)
             set_property(CACHE CMAKE_BUILD_TYPE
                     PROPERTY STRINGS ${CMAKE_CONFIGURATION_TYPES})
         endif()
-
-        # RelWithDebInfo uses -O2 which is sub-ideal for how it is intended to be used
-        #   https://gitlab.kitware.com/cmake/cmake/-/merge_requests/591
-        list(TRANSFORM CMAKE_C_FLAGS_RELWITHDEBINFO REPLACE "-O2" "-O3")
-        list(TRANSFORM CMAKE_CXX_FLAGS_RELWITHDEBINFO REPLACE "-O2" "-O3")
-        # FIXME: due to the "AMReX inits CUDA first" logic we will first see this with -O2 in output
-        list(TRANSFORM CMAKE_CUDA_FLAGS_RELWITHDEBINFO REPLACE "-O2" "-O3")
```

Comment:
This line was called before we set `enable_language(cuda)`.
The result was an empty CUDA flag list, which as a result compiled the host with `-O0` (GCC) and the device with `-O3` (NVCC) for `CUDA` language translation units (TUs):
```
nvcc [no -O flags] [includes] [file].cpp
```

When calling `cmake -S <src_dir>` or `ccmake` again on the build directory, the CUDA language was enabled and then this was initialized correctly to `-O3 ...` for `nvcc`-compiled TUs.